core/src/main/java/com/linecorp/armeria/common/stream/SubscriptionOption.java

```diff
@@ -19,27 +19,22 @@
 import org.reactivestreams.Subscriber;
 import org.reactivestreams.Subscription;
 
-import io.netty.buffer.ByteBuf;
-import io.netty.buffer.ByteBufHolder;
+import com.linecorp.armeria.internal.stream.InternalSubscriptionOption;
+
 import io.netty.util.concurrent.EventExecutor;
 
 /**
- * Options used when subscribing to a {@link StreamMessage}.
+ * Options used when subscribing to a {@link StreamMessage}. This class is sealed to Armeria and can only be
+ * implemented here.
  *
  * @see StreamMessage#subscribe(Subscriber, SubscriptionOption...)
  * @see StreamMessage#subscribe(Subscriber, EventExecutor, SubscriptionOption...)
  */
-public enum SubscriptionOption {
-
-    /**
-     * To receive the pooled {@link ByteBuf} and {@link ByteBufHolder} as is, without making a copy.
-     * If you don't know what this means, do not specify this when you subscribe the {@link StreamMessage}.
-     */
-    WITH_POOLED_OBJECTS,
+public interface SubscriptionOption {
```

Comment:
Since the original options were effectively booleans (we only compare equality with an enum) I figured it's ok to use this pattern which replicates it. `AbstractOption` has a lot of complexity because it supports arbitrary types, but I don't think we need it here, it would just make the code in general more verbose / slower.

Reply:
Ah, yeah. Forgot about the value part. 😅